dietpi/dietpi-software

```diff
@@ -11828,7 +11831,10 @@ Description=Jackett Daemon
 After=network.target
 
 [Service]
-User=root
+User=jackett
+Group=jackett
+WorkingDirectory=/opt/jackett
+Environment=USER=jackett HOME=/opt/jackett
```

Comment:
This trick will make believe **mono** that home folder is **/opt/jackett**, it will create to folders: [.mono;.config] in **/opt/jackett**. Easy and convenient to keep all files in one folder.

Another option will be creating folder **/etc/jackett** and make **mono** think that this is home folder, no big difference but it will be more close to linux conventions.

Let me know what do you think ...

Reply:
Great work. Yeah I like having all related files in one place instead of spread around the system. This also makes backups and migrations from one to another system easier 👍.

`/opt/jackett` vs `/etc/jackett` is as far as I read somewhere non-package software vs packaged software. As Jackett is not  installed via APT/deb package, I think /opt fits better. However e.g. we had this discussion once when moving DietPi config/related files from `/etc/dietpi`. But we moved it to `/var/lib/dietpi` instead of `/opt/dietpi`, but to be honest, I don't remember exactly which one fit's better and why. Also ideas about which one is "correct" diverge 😉.

€: Ah /var more for variable files, e.g. /var/lib/mysql as database folder, /var/lib/dietpi/dietpi-ramlog for persistent log storage and such, not so much for config files. Config files inside /etc or optionally for "optional" software in /opt.
Thus I think /opt is just fine.

Reply:
Hm, 🤔 interesting... 
What do you think then about migration of **gogs** from `/etc/gogs` to `/opt/gogs` ?
It's a single folder app with binary executable configs and so on ... (except repo folder of sure)
Should I open one more PR 😁 ?

Reply:
Good question, I am just reading a bid more about this: https://unix.stackexchange.com/questions/11544/what-is-the-difference-between-opt-and-usr-local

The question really is, if we want to follow the FHS standard, or if we want to safe/install our manual (non-package) software installations in a monolithic way for easier configuration/backup/migration etc.
Also for software placed in `/opt` it is according to FHS standard still just `/opt/package` for binaries/libraries (if I am not mistaken) of "optional" software packages, while their configuration files should be then places within `/etc/opt/package` and variable files (databases, temporary, logs, working dir) in `/var/opt/package`, so very non-monolithic.

I don't know, but I would like to have just config and variable files inside a separate folder, just everything that you need to backup or edit/configure. Binaries + libraries doesn't matter, since you can reinstall them at any time.

Maybe @Fourdee has an opinion about that. I would actually like to handle this similar for all our offered software, where we have control over their install/working/config directories.

But all this is more a minor one. In my opinion worth to discuss but not yet ready to start reworking software titles just for this, as it works very well as it is 😉.

Reply:
Thanks for the link 👍.  (I have to read more about FHS 😉)

Regarding backups...
My personal preference is to have symlinks for all data/config folders and files.
I keep real data in `/mnt/dietpi_userdata/.persistent` after fresh installing anything I have a script that just replaces default config/data with the symlink to my ".persistent" folder and restart the app.
In this way, backups are extremely lightweight and I have no fear of installing from scratch the OS.
The downside is that you have to know where each app is keeping it's data and configs folders, however, I didn't found this too complex so far...
This approach was also very helpful on migration from deprecated ~v159 to new v6+ where backups where incompatible.